extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java

```diff
@@ -424,6 +428,32 @@ public void registerRoutes(WebSocketServerRecorder recorder, HttpRootPathBuildIt
         }
     }
 
+    @BuildStep
+    UnremovableBeanBuildItem makeHttpUpgradePoliciesUnremovable() {
+        // we access the policies programmatically
+        return UnremovableBeanBuildItem.beanTypes(HTTP_UPGRADE_POLICY_NAME);
+    }
+
+    @BuildStep
+    List<ValidationPhaseBuildItem.ValidationErrorBuildItem> validateHttpUpgradePolicyNotRequestScoped(
+            ValidationPhaseBuildItem validationPhase) {
+        return validationPhase
+                .getContext()
```

Comment:
You could use:
```java
getContext()
   .beans()
   .withBeanType(HTTP_UPGRADE_POLICY_NAME)
   .filter(b -> {
                    var targetScope = BuiltinScope.from(b.getScope().getDotName());
                    return BuiltinScope.APPLICATION != targetScope
                            && BuiltinScope.SINGLETON != targetScope
                            && BuiltinScope.DEPENDENT != targetScope;
                })...
```  

We could even add `BeanStream#withAnyScope(...)` or so...

Reply:
I am not able to assess if someone else needs it, but I'd not like to add in this PR anything but WS Next because it's really beautiful how quick CI runs for this PR.

Reply:
I totally get it ;-)